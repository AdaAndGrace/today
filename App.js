import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import {Icon} from 'react-native-elements';
import TodoList from './src/components/TodoList/TodoList';
import { Font } from 'expo';
import TodayText from './src/components/TodayText/TodayText'


export default class App extends React.Component {

    state = {
        fontLoaded: false,
        list: [
            {
                title: 'Take out the trash',
                done: true
            },
            {
                title: 'Buy Coffee',
                done: false
            },
            {
                title: 'Call Amy',
                done: false
            }
        ]
    };
    async componentDidMount() {
        await Font.loadAsync({
            'WaitingForTheSunrise': require('./src/assets/fonts/WaitingForTheSunrise.ttf'),
        });

        this.setState({ fontLoaded: true });
    }


    handleAddItem = (item) => {
        console.log(item)
    };

    handleCheckItem = (index) => {
        let list = this.state.list;
        list[index].done = !this.state.list[index].done;
        this.setState({
            list: list
        });
    };


    render() {
        return (

            <React.Fragment>
                {this.state.fontLoaded ? (
                    <View style={styles.container}>
                        <TodayText h2 style={{fontSize: 56}}>Today!</TodayText>


                        <TodoList items={this.state.list} onCheckItem={this.handleCheckItem}/>
                        <Icon
                            name='add-circle'
                            color='#f50'
                            onPress={() => this.handleAddItem('TODO: pop a modal to capture new item.  also, style this button plz')}/>
                    </View>
                    ) : null
                }
            </React.Fragment>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'flex-start',
        marginTop: 40,
    },
    list: {
        width: 200
    },
    listItem: {
        paddingTop: 0,
        paddingRight: 0,
        paddingBottom: 0,
        paddingLeft: 0,
        margin: 0
    },
    listCheckbox: {
        backgroundColor: 'transparent',
        borderWidth: 0,
        marginTop: 0,
        marginRight: 0,
        marginBottom: 0,
        marginLeft:0
    },
    itemText: {
        fontSize: 16
    },
    doneItemText: {
        color: '#aaa',
        fontSize: 16,
        textDecorationLine: 'line-through'
    }
});
