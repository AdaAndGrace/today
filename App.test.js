import React from 'react';
import App from './App';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const wrapper = renderer.create(<App/>).toJSON();
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
});