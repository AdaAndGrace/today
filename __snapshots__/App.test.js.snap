// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "marginTop": 40,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        undefined,
        Object {
          "fontSize": 42.5,
        },
        undefined,
        undefined,
        undefined,
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Today!
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#bbb",
          "borderTopWidth": 1,
          "marginTop": 20,
        },
        Object {
          "width": 350,
        },
      ]
    }
  >
    <View
      disabled={false}
      onLongPress={undefined}
      onPress={undefined}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "margin": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          false,
          false,
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={undefined}
          >
            <View>
              <View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#ededed",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "margin": 5,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        false,
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      onLongPress={undefined}
                      onPress={undefined}
                      style={
                        Array [
                          Object {
                            "color": "green",
                            "fontSize": 30,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      ÔÅò
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "#43484d",
                              "fontWeight": "bold",
                              "marginLeft": 10,
                              "marginRight": 10,
                            },
                            Object {
                              "color": "#aaa",
                              "fontSize": 16,
                              "textDecorationLine": "line-through",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      Take out the trash
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={undefined}
          >
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      disabled={false}
      onLongPress={undefined}
      onPress={undefined}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "margin": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          false,
          false,
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={undefined}
          >
            <View>
              <View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#ededed",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "margin": 5,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        false,
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      onLongPress={undefined}
                      onPress={undefined}
                      style={
                        Array [
                          Object {
                            "color": "#bfbfbf",
                            "fontSize": 30,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      Ôáõ
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "#43484d",
                              "fontWeight": "bold",
                              "marginLeft": 10,
                              "marginRight": 10,
                            },
                            Object {
                              "fontSize": 16,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      Buy Coffee
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={undefined}
          >
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      disabled={false}
      onLongPress={undefined}
      onPress={undefined}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "#bbb",
            "borderBottomWidth": 1,
            "paddingBottom": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
          Object {
            "margin": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          false,
          false,
        ]
      }
      underlayColor="white"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginLeft": 10,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={undefined}
          >
            <View>
              <View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "#ededed",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "margin": 5,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                        },
                        false,
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={false}
                      ellipsizeMode="tail"
                      onLongPress={undefined}
                      onPress={undefined}
                      style={
                        Array [
                          Object {
                            "color": "#bfbfbf",
                            "fontSize": 30,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                        ]
                      }
                    >
                      Ôáõ
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "#43484d",
                              "fontWeight": "bold",
                              "marginLeft": 10,
                              "marginRight": 10,
                            },
                            Object {
                              "fontSize": 16,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      Call Amy
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={undefined}
          >
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        false,
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#f50",
            "fontSize": 24,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ÓÖá
    </Text>
  </View>
</View>
`;
